dl4j-cv-labs/src/main/java/ai/certifai/utilities/VocLabelProvider.java

```diff
@@ -0,0 +1,115 @@
+package ai.certifai.utilities;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.NonNull;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.datavec.image.recordreader.objdetect.ImageObject;
+import org.datavec.image.recordreader.objdetect.ImageObjectLabelProvider;
+
+public class VocLabelProvider implements ImageObjectLabelProvider {
+    private static final String OBJECT_START_TAG = "<object>";
+    private static final String OBJECT_END_TAG = "</object>";
+    private static final String NAME_TAG = "<name>";
+    private static final String XMIN_TAG = "<xmin>";
+    private static final String YMIN_TAG = "<ymin>";
+    private static final String XMAX_TAG = "<xmax>";
+    private static final String YMAX_TAG = "<ymax>";
+    private String annotationsDir;
+
+    public VocLabelProvider(@NonNull String baseDirectory) {
+        if (baseDirectory == null) {
+            throw new NullPointerException("baseDirectory is marked non-null but is null");
```

Comment:
is this redundant given the @NonNull annotation?

Reply:
hmm..
I did not look at it carefully when I was doing this...
true, this is redundant. Will remove the if statement..